packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -417,15 +417,20 @@ export function calculatePositionInternal(
 
   // All values are transformed so that 0 is at the top/left of the overlay depending on the orientation
   // Prefer the arrow being in the center of the trigger/overlay anchor element
-  let preferredArrowPosition = childOffset[crossAxis] + .5 * childOffset[crossSize] - position[crossAxis];
+  // childOffset[crossAxis] + .5 * childOffset[crossSize] = absolute position with respect to the trigger's coordinate system that would place the arrow in the center of the trigger
+  // position[crossAxis] - margins[AXIS[crossAxis]] = value use to transform the position to a value with respect to the overlay's coordinate system. A child element's (aka arrow) position absolute's "0"
+  // is positioned after the margin of its parent (aka overlay) so we need to subtract it to get the proper coordinate transform
+  let preferredArrowPosition = childOffset[crossAxis] + .5 * childOffset[crossSize] - position[crossAxis] - margins[AXIS[crossAxis]];
 
   // Min/Max position limits for the arrow with respect to the overlay
   const arrowMinPosition = arrowSize / 2 + arrowBoundaryOffset;
-  const arrowMaxPosition = overlaySize[crossSize] - (arrowSize / 2) - arrowBoundaryOffset;
+  // overlaySize[crossSize] - margins['left'] - margins['right'] = true size of the overlay
+  const arrowMaxPosition = (overlaySize[crossSize] - margins['left'] - margins['right']) - (arrowSize / 2) - arrowBoundaryOffset ;
```

Comment:
Bit wary about this since it assumes the margins shouldn't be considered part of the physical overlay (aka like S2 tooltip). Not sure about complications from negative margins too...

Reply:
should we be using a cross offset instead?

Reply:
ah yep thanks, forgot to convert these as well